src/libPMacc/include/cuSTL/algorithm/kernel/Reduce.tpp

```diff
@@ -36,124 +37,74 @@ namespace algorithm
 {
 namespace kernel
 {
-
+    
 namespace detail
 {
-
-template<typename BlockDim, int dim>
-struct ReduceKernel
+    
+/**
+ * @class MappedNavigator
+ * @author Heiko Burau 
+ * @date 30/06/15
+ * @file Reduce.tpp
+ * @brief Wraps a ndim-cursor into a 1D-cursor
+ */
+template<int T_dim>
+class MappedNavigator
```

Comment:
Please rename this file to `MappedNavigator.tpp`. We name files like the class inside.